src/components/DatePicker/index.android.js

```diff
@@ -14,50 +15,59 @@ class DatePicker extends React.Component {
         };
 
         this.showPicker = this.showPicker.bind(this);
-        this.raiseDateChange = this.raiseDateChange.bind(this);
-    }
-
-    /**
-     * @param {Event} event
-     */
-    showPicker(event) {
-        this.setState({isPickerVisible: true});
-        event.preventDefault();
+        this.setDate = this.setDate.bind(this);
     }
 
     /**
      * @param {Event} event
      * @param {Date} selectedDate
      */
-    raiseDateChange(event, selectedDate) {
+    setDate(event, selectedDate) {
         if (event.type === 'set') {
-            this.props.onChange(selectedDate);
+            this.props.onInputChange(selectedDate);
         }
 
         this.setState({isPickerVisible: false});
     }
 
+    /**
+     * @param {Event} event
+     */
+    showPicker(event) {
+        this.setState({isPickerVisible: true});
+        event.preventDefault();
+    }
```

Comment:
There was a lint warning that `showPicker` should come after `setDate`, that's why I moved this.